src/node_url_pattern.h

```diff
@@ -0,0 +1,104 @@
+#ifndef SRC_NODE_URL_PATTERN_H_
+#define SRC_NODE_URL_PATTERN_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "ada.h"
+#include "base_object.h"
+#include "util.h"
+
+#include <v8.h>
+
+#include <optional>
+#include <string_view>
+
+namespace node::url_pattern {
+
+class URLPatternRegexProvider {
```

Comment:
Nit: Some code comments explaining what this is would be helpful for others coming into this code later